app/src/main/java/com/duckduckgo/app/browser/favorites/FocusedLegacyViewModel.kt

```diff
@@ -0,0 +1,217 @@
+import com.duckduckgo.di.scopes.ViewScope
+import com.duckduckgo.savedsites.api.SavedSitesRepository
+import com.duckduckgo.savedsites.api.models.BookmarkFolder
+import com.duckduckgo.savedsites.api.models.SavedSite
+import com.duckduckgo.savedsites.api.models.SavedSite.Bookmark
+import com.duckduckgo.savedsites.api.models.SavedSite.Favorite
+import com.duckduckgo.sync.impl.ui.SyncActivityViewModel.ViewState
+import javax.inject.Inject
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+@SuppressLint("NoLifecycleObserver") // we don't observe app lifecycle
+@ContributesViewModel(ViewScope::class)
+class FocusedLegacyViewModel @Inject constructor(
+    private val dispatchers: DispatcherProvider,
+    private val savedSitesRepository: SavedSitesRepository,
+) : ViewModel(), DefaultLifecycleObserver, EditSavedSiteListener, DeleteBookmarkListener {
+
+    data class ViewState(
+        val favourites: List<Favorite> = emptyList(),
+    )
+
+    sealed class Command {
+        class ShowEditSavedSiteDialog(val savedSiteChangedViewState: SavedSiteChangedViewState) : Command()
+        class DeleteFavoriteConfirmation(val savedSite: SavedSite) : Command()
+        class DeleteSavedSiteConfirmation(val savedSite: SavedSite) : Command()
+    }
+
+    val hiddenIds = MutableStateFlow(HiddenBookmarksIds())
+
+    private val _viewState = MutableStateFlow(ViewState())
+    val viewState = _viewState.asStateFlow()
+    private val command = Channel<Command>(1, BufferOverflow.DROP_OLDEST)
+    internal fun commands(): Flow<Command> = command.receiveAsFlow()
+
+    override fun onStart(owner: LifecycleOwner) {
+        super.onStart(owner)
+
+        viewModelScope.launch(dispatchers.io()) {
+            savedSitesRepository.getFavorites()
+                .combine(hiddenIds) { favorites, hiddenIds ->
+                    favorites.filter { it.id !in hiddenIds.favorites }
+                }
+                .flowOn(dispatchers.io())
+                .onEach { favourites ->
+                    withContext(dispatchers.main()) {
+                        _viewState.emit(
+                            viewState.value.copy(
+                                favourites = favourites,
+                            ),
+                        )
+                    }
+                }
+                .flowOn(dispatchers.main())
+                .launchIn(viewModelScope)
+        }
+    }
+
+    fun onEditSavedSiteRequested(savedSite: SavedSite) {
```

Comment:
When I edit a favorite, doesn't update in the new tab. Not sure if new but do we know why?

Reply:
+1